src/styles/main.scss

```diff
@@ -0,0 +1,65 @@
+@import './utils/variables';
+
+body {
+  margin: 0;
+  padding-top: 25px;
+}
+
+.calendar {
+  display: flex;
+  gap: $gap;
+  flex-wrap: wrap;
+  width: $col-count * $cell-width + $gap * $col-count;
+  margin: 0 auto;
+
+  &__day {
+    display: flex;
+    font-family: Arial, Helvetica, sans-serif;
+    justify-content: center;
+    align-items: center;
+    font-size: 30px;
+    line-height: 100%;
+    box-sizing: border-box;
+    background-color: #eee;
+    width: $cell-width;
+    height: $cell-height;
+    border: 1px solid black;
+
+    &:first-child {
+      margin-left: ($col-count - 1) * $cell-width + $gap * ($col-count - 1);
+    }
+
+    @each $day, $number in $days-of-week {
+      @if $day == $current-day {
+        &:first-child {
+          margin-left: $number * $cell-width + $gap * $number;
+        }
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: #ffbfcb;
+      cursor: pointer;
+      transition-duration: 500ms;
+    }
+
+    @for $days from 1 through 31 {
+      &:nth-child(#{$days})::before {
+        content: '#{$days}';
+      }
+    }
+
+    @for $i from $month-length + 1 through 32 {
```

Comment:
The loop for hiding days beyond the `$month-length` should be verified. Ensure that `$month-length` is set correctly to reflect the actual number of days in the month.